app/vmalert/main.go

```diff
@@ -39,45 +40,88 @@ absolute path to all .yaml files in root.`)
 	externalURL              = flag.String("external.url", "", "External URL is used as alert's source for sent alerts to the notifier")
 )
 
-// TODO: hot configuration reload
 // TODO: alerts state persistence
 func main() {
 	envflag.Parse()
 	buildinfo.Init()
 	logger.Init()
 	checkFlags()
-	ctx, cancel := context.WithCancel(context.Background())
 	eu, err := getExternalURL(*externalURL, *httpListenAddr, httpserver.IsTLS())
 	if err != nil {
 		logger.Fatalf("can not get external url:%s ", err)
 	}
 	notifier.InitTemplateFunc(eu)
 
-	logger.Infof("reading alert rules configuration file from %s", strings.Join(*rulePath, ";"))
+	wg := sync.WaitGroup{}
```

Comment:
as vmalert stores alerts in memory in Rule struct, all alerts will be lost during config reload, we need to thing about how to avoid it

Reply:
The alert state is stored per-group per-rule. Probably, we can check for sure what groups/rules were changed and reload only them.

Reply:
emmm, good question

Reply:
I will make more change to do per group/ per rule reload, as now ctx is shared for all groups, cancel() need to do more change to make this work